listenbrainz/webserver/static/js/src/recommendations/Recommendations.tsx

```diff
@@ -62,6 +65,76 @@ export default class Recommendations extends React.Component<
     );
   }
 
+  componentDidMount(): void {
+    const { user, currentUser } = this.props;
+    const { currRecPage } = this.state;
+    if (currentUser?.name === user?.name) {
+      this.loadFeedback();
+    }
+    window.history.replaceState(null, "", `?page=${currRecPage}`);
+  }
+
+  getFeedback = async () => {
+    const { user, recommendations } = this.props;
+    const recordings: string[] = [];
+
+    if (recommendations) {
+      recommendations.forEach((recommendation) => {
+        const recordingMbid = get(
+          recommendation,
+          "track_metadata.additional_info.recording_mbid"
+        );
+        if (recordingMbid) {
+          recordings.push(recordingMbid);
+        }
+      });
+      try {
+        const data = await this.APIService.getFeedbackForUserForRecommendations(
+          user.name,
+          recordings.join(",")
+        );
+        return data.feedback;
+      } catch (error) {
+        this.newAlert(
+          "danger",
+          "Playback error",
+          typeof error === "object" ? error.message : error
+        );
+      }
+    }
+    return {};
```

Comment:
Found another small bug:
```suggestion
    return [];
```
Since the API returns an array, and we're using `feedback.forEach(` in the `loadFeedback` function (`forEach` is a method of array, no of objects), when there is no feedback currently an error will be thrown: `TypeError: feedback.forEach is not a function`